### PR TITLE
smt: Rename NodeUpdate to Node

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change renames `NodeUpdate` to a more generic `Node` type, because
it is used both for updates and reads.

### CODE REVIEW

File: merkle/smt/hstar3.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type HashChildrenFn func(left, right []byte) []byte
 //
 // TODO(pavelkalinnikov): Swap in the code, and document it properly.
 type HStar3 struct {
-	upd   []NodeUpdate
+	upd   []Node
```
Comment (Martin2112) [inline merkle/smt/hstar3.go]: The name `upd` is now not quite as relevant. Maybe `nodes` or similar is better.

File: merkle/smt/nodes.go [hunk 2]
```diff
@@ -22,28 +22,28 @@ import (
...
 //
-// TODO(pavelkalinnikov): Make this algorithm independent of NodeUpdate type.
-func Prepare(updates []NodeUpdate, depth uint) error {
-	for i := range updates {
-		if d, want := updates[i].ID.BitLen(), depth; d != want {
+// TODO(pavelkalinnikov): Make this algorithm independent of Node type.
+func Prepare(nodes []Node, depth uint) error {
+	for i := range nodes {
+		if d, want := nodes[i].ID.BitLen(), depth; d != want {
 			return fmt.Errorf("upd #%d: invalid depth %d, want %d", i, d, want)
```
Comment (Martin2112) [inline merkle/smt/nodes.go]: `upd` -> `node` probably.

File: merkle/smt/nodes_test.go [hunk 3]
```diff
@@ -30,22 +30,22 @@ func TestPrepare(t *testing.T) {
...
-		upd     []NodeUpdate
-		want    []NodeUpdate
+		upd     []Node
+		want    []Node
 		wantErr string
 	}{
-		{desc: "depth-err", upd: []NodeUpdate{{ID: id1.Prefix(10)}}, wantErr: "invalid depth"},
-		{desc: "dup-err1", upd: []NodeUpdate{{ID: id1}, {ID: id1}}, wantErr: "duplicate ID"},
-		{desc: "dup-err2", upd: []NodeUpdate{{ID: id1}, {ID: id2}, {ID: id1}}, wantErr: "duplicate ID"},
+		{desc: "depth-err", upd: []Node{{ID: id1.Prefix(10)}}, wantErr: "invalid depth"},
```
Comment (Martin2112) [inline merkle/smt/nodes_test.go]: Not sure it's worth renaming these `upd`s

File: merkle/smt/writer.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ type NodeBatchAccessor interface {
 	// The returned hashes may be missing or be nil for empty subtrees.
 	Get(ctx context.Context, ids []tree.NodeID2) (map[tree.NodeID2][]byte, error)
 	// Set applies the given node hash updates.
-	Set(ctx context.Context, upd []NodeUpdate) error
+	Set(ctx context.Context, upd []Node) error
```
Comment (Martin2112) [inline merkle/smt/writer.go]: This `upd` and the one below in `Split` is a possible for rename but not essential.

File: merkle/smt/writer_test.go [hunk 5]
```diff
@@ -240,12 +240,12 @@ func update(ctx context.Context, t testing.TB, w *Writer, acc NodeBatchAccessor,
 	return rootUpd
 }
 
-// genUpd returns a NodeUpdate for the given key and value. The returned node
-// ID is 256-bit map key based on SHA256 of the given key string.
-func genUpd(key, value string) NodeUpdate {
+// genUpd returns a Node for the given key and value. The returned node ID is
```
Comment (Martin2112) [inline merkle/smt/writer_test.go]: Not sure it's worth renaming this or other test related occurrences. Up to you.

File: server/map_rpc_server.go [hunk 6]
```diff
@@ -336,10 +336,10 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 
 	// Overwrite/set the leaf hashes in the request and create a summary of
 	// the leaf indices and new hash values.
-	upd := make([]smt.NodeUpdate, 0, len(req.Leaves))
+	upd := make([]smt.Node, 0, len(req.Leaves))
```
Comment (Martin2112) [inline server/map_rpc_server.go]: Maybe change this to `nodes`.

File: server/map_tree_updater.go [hunk 7]
```diff
@@ -39,7 +39,7 @@ type mapTreeUpdater struct {
 // update updates the sparse Merkle tree at the passed-in revision with the
 // given leaf updates, and writes it to the storage. Returns the new root hash.
 // Requires updates to be non-empty.
-func (t *mapTreeUpdater) update(ctx context.Context, tx storage.MapTreeTX, upd []smt.NodeUpdate, writeRev int64) ([]byte, error) {
+func (t *mapTreeUpdater) update(ctx context.Context, tx storage.MapTreeTX, upd []smt.Node, writeRev int64) ([]byte, error) {
```
Comment (Martin2112) [inline server/map_tree_updater.go]: `nodes` possibly clearer but not an essential rename.

**Comment by pavelkalinnikov** [comment]
Renamed all the `upd` occurrences.

**Pull request merged**